docs/cbuild-run-enhancement.md

```diff
@@ -0,0 +1,129 @@
+&nbsp;&nbsp;&nbsp; `ram-size:`                    | uint   | **Required** | Maximum size of RAM available for the execution of the Flash programming algorithm.
+&nbsp;&nbsp;&nbsp; `default:`                     | bool   |   Optional   | If `true`, then this is the default Flash programming algorithm that gets configured in a project. Default value is `false`.
+&nbsp;&nbsp;&nbsp; `pname:`                       | string |   Optional   | Processor identifier. Mandatory for devices that embed multiple processors that require different algorithms.
+
+
+### `system-descriptions:`
+`system-descriptions:`                            | Value  |   Use        | Content
+:-------------------------------------------------|--------|--------------|:------------------------------------
+`- file:`                                         | string | **Required** | Specifies the file name including the path.
+&nbsp;&nbsp;&nbsp; `type:`                        | string | **Required** | Specifies the file type. Predefined values: `svd`, `sdf`.
+&nbsp;&nbsp;&nbsp; `info:`                        | string |   Optional   | Brief description of the file.
+
+
+### `output:`
+`output:`                                         | Value  |   Use        | Content
+:-------------------------------------------------|--------|--------------|:------------------------------------
+`- file:`                                         | string | **Required** | Specifies the file name including the path.
+&nbsp;&nbsp;&nbsp; `type:`                        | string | **Required** | Specifies the file type. Predefined values: `lib`, `elf`, `hex`, `bin`, `map`.
+&nbsp;&nbsp;&nbsp; `info:`                        | string |   Optional   | Brief description of the file.
+&nbsp;&nbsp;&nbsp; `run:`                         | string |   Optional   | Additional command string for download or programming.
+&nbsp;&nbsp;&nbsp; `debug:`                       | string |   Optional   | Additional command string for debug.
+> Where do `run` and `debug` come from?
+
+### `system-resources:`
+`system-resources:`                               | Value  |   Use        | Content
+:-------------------------------------------------|--------|--------------|:------------------------------------
+&nbsp;&nbsp;&nbsp; `memory:`                      |        |   Optional   | Identifies the section for memory.
+
+`memory:`                                         | Value  |   Use        | Content
+:-------------------------------------------------|--------|--------------|:------------------------------------
+`- name:`                                         | string | **Required** | Name of the memory region (when PDSC contains id, it uses the id as name).
+&nbsp;&nbsp;&nbsp; `access:`                      | string | **Required** | Access permission of the memory (combination of `r`, `w`, `x`, `p`, `s`, `n`, `c`).

```

Comment:
While I think that the cbuild-run file generator should always write this to support debuggers, one could argue `optional` + a well-defined default is enough. But let's probably start with `required`. It's easier to relax this in future than the other way around.